### PR TITLE
MDM Micronode + Lists

### PR DESCRIPTION
**Pull request by plyhun**
## Abstract

The content micronode + list fields implementation.

### General

* [ ] Added abstract that describes the change
* [ ] Added changelog entry to `/CHANGELOG.adoc`
* [ ] Ensured that the change is covered by tests
* [ ] Ensured that the change is documented in the docs

### On API Changes

* [ ] Checked if the changes are breaking or not
* [ ] Added GraphQL API if applicable
* [ ] Added Elasticsearch mapping if applicable

### CODE REVIEW

File: mdm/api/src/main/java/com/gentics/mesh/core/data/dao/NodeDao.java [hunk 1]
```diff
@@ -475,15 +481,4 @@
 	 * @return
 	 */
 	NodeReference transformToReference(HibNode node, InternalActionContext ac);
```
Comment (plyhun) [inline mdm/api/src/main/java/com/gentics/mesh/core/data/dao/NodeDao.java:483]: The duplicate removed. As the function is called from the webroot context, I have left the `getWebrootEdges` name, and renamed Umberto's implementation.

File: mdm/api/src/main/java/com/gentics/mesh/core/data/node/HibMicronode.java [hunk 2]
```diff
@@ -39,22 +46,38 @@
 	 */
 	Result<? extends HibNodeFieldContainer> getContainers();
```
Comment (plyhun) [inline mdm/api/src/main/java/com/gentics/mesh/core/data/node/HibMicronode.java:48]: Dup removed.

File: mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingContentDao.java [hunk 3]
```diff
@@ -115,11 +116,63 @@ default void migrateContainerOntoBranch(HibNodeFieldContainer container, HibBran
...
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Find all micronodes. Used by the indexing engine and tests.
 	 * 
 	 * @return
 	 */
```
Comment (plyhun) [inline mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingContentDao.java:174]: `getMicronodePersistenceClass` is not needed anymore.

File: mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingNodeDao.java [hunk 4]
```diff
@@ -1719,78 +1716,43 @@ default Path resolvePath(HibNode baseNode, String branchUuid, ContainerType type
 		return path;
 	}
```
Comment (plyhun) [inline mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingNodeDao.java:1718]: Moved into PersistingContentDao for consistency.

File: mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingSchemaDao.java [hunk 5]
```diff
@@ -259,26 +257,6 @@ default HibSchema create(SchemaVersionModel schema, HibUser creator, String uuid
 				&& userDao.hasPermissionForId(user, node.getId(), READ_PUBLISHED_PERM)));
 	}
```
Comment (plyhun) [inline mdm/common/src/main/java/com/gentics/mesh/core/data/dao/PersistingSchemaDao.java:259]: Dups removed.

**Review by udovidio — APPROVED** [review]

**Review by npomaroli — APPROVED** [review]

**Pull request merged**